dev/ColorPicker/readme.md

```diff
@@ -0,0 +1,141 @@
+
+## SpectrumBrush
+
+Provides a way of drawing the `ColorSpectrum` background using the composition APIs in Windows. Internally, this uses two `CompositionSurfaceBrush` and both a minimum and maximum surface composited together. This brush is how the spectrum is rendered to the display on Windows version 1703 (RS2) and later.  
+
+## Ideas for Future Implementations / Lessons Learned
+
+* RGB and HSV channels are identified separately in most places. For example, HSV channels are represented by the `ColorPickerHsvChannel` enum. Since HSV is used as the primary representation internally there is no corresponding enum for RGB channels. In the control template, however, there are separate controls depending on the active color representation RGB/HSV. This means there is a duplication of a large number of controls. These could all be unified, for example, by referring to both hue and red as channel 1, etc. This greatly simplifies the effort required in the control template itself and removes duplicate controls. Instead, the same input controls can be used and only the active color representation set. Internal representation could add an enum for Channel1, Channel2, Channel3 and Alpha.
+
+* The 'ThirdDimension' slider really should not be named 'dimension'. 'Channel' is the standardized term so 'ThirdChannelSlider' would be a better choice. 'Components' is also the term used in control properties.
+
+* HSV color representation should be externally exposed by the color picker for those applications that require higher levels of precision. A new HsvColor struct should be added to the existing RGB struct already in WinUI.
```

Comment:
Different opinion, that's fine. This is a roll-up of lessons learned various places and hopefully if the ColorPickerButton ideas from the community toolkit ever come back to this repo changes like this can be made.

Reply:
And why not make this change regardless of the WCT ColorPickerButton? This is a feature gap and just saying "yeah this was a mistake" and not fixing this seems a bit strange in my opinion.

Reply:
I'm inclined to agree with @chingucoding - a readme should document existing design and behavior; if there's additional bug fixes or feature requests, those should be filed as issues rather than being put here.

Reply:
If I'm in the minority it can be changed. I just have a differing opinion and didn't want to change that without a majority consensus.  That said, I think there are two types of topics in this section: feature additions (like HsvColor being external) and architecture changes. The feature additions probably should be separated into different issues. Architecture changes to the control though still seem like a better fit in this document.

Reply:
What I've done is separated features from implementation/architectural changes. Architectural changes should still live in this file as I don't see them changing for a long time if-ever. However, if ColorPicker is ever re-written it would be good to make some changes.

This means the HsvColor Property is now a feature-request #3281